src/GraphQL/Utilities/AstPrinter.cs

```diff
@@ -328,25 +336,54 @@ public AstPrintVisitor()
                 c.Field(x => x.Value);
                 c.Print(f =>
                 {
-                    var val = f.Arg(x => x.Value);
-                    if (!string.IsNullOrWhiteSpace(val?.ToString()) && !val.ToString().StartsWith("\"", StringComparison.InvariantCulture))
+                    var val = (string)f.Arg(x => x.Value);
+                    if (val == null)
+                        throw new InvalidOperationException("StringValue cannot contain null");
+                    var sb = new System.Text.StringBuilder(val.Length + 2);
+                    sb.Append('\"');
+                    foreach (var ch in val)
                     {
-                        val = $"\"{val}\"";
+                        if (ch < ' ')
+                        {
+                            if (ch == '\n')
+                                sb.Append("\\\n");
+                            else if (ch == '\r')
+                                sb.Append("\\\r");
+                            else if (ch == '\t')
+                                sb.Append("\\\t");
+                            else
+                                sb.Append("\\u" + ((int)ch).ToString("X4"));
+                        }
+                        else if (ch == '\\')
+                            sb.Append("\\\\");
+                        else if (ch == '\"')
+                            sb.Append("\\\"");
+                        else
+                            sb.Append(ch);
                     }
-                    return val;
+                    sb.Append('\"');
+                    return sb;
                 });
+
             });
 
             Config<BooleanValue>(c =>
             {
                 c.Field(x => x.Value);
-                c.Print(f => f.Arg(x => x.Value)?.ToString().ToLower(CultureInfo.InvariantCulture));
+                c.Print(f => (bool)f.Arg(x => x.Value) ? "true" : "false");
             });
 
             Config<EnumValue>(c =>
             {
                 c.Field(x => x.Name);
-                c.Print(p => p.Arg(x => x.Name));
+                c.Print(p =>
+                {
+                    var name = (string)p.Arg(x => x.Name);
+                    if (name == null)
+                        throw new InvalidOperationException("EnumValue name cannot be null");
+                    NameValidator.ValidateDefault(name, NamedElement.EnumValue);
```

Comment:
I doubt that this check is required here. You need to print what is already there. It is possible that someone needs to print values like `KEBAB-ENUM`. Also I would move null check to `EnumValue` ctor from here.

Reply:
And possible add null check into `NameNode` ctor.

Reply:
Any enum value with invalid characters cannot be printed, as it does not meet the graphql language syntax.  What if the character had a curly brace in it?  Correct me if I'm wrong, but this ASTPrinter should print a string that conforms to the GraphQL syntax.  Allowing malformed documents to be created opens up security holes similar to the classic SQL injection attack.

Reply:
Let's change this assertion `Any enum value with invalid characters cannot be printed` a bit -> `Any enum value with invalid characters cannot be created` . I suggest to move check inside `EnuValue` ctor.

Reply:
I made changes myself not to delay the release.